docs/extensions/secure/execution-types.md

```diff
@@ -0,0 +1,403 @@
+---
+description: Learn about the permissions and execution types available in Directus Secure Extensions
+contributors: Nils Twelker, Kevin Lewis, Esther Agbaje
+---
+
+# Execution Types
+
+In a Secure Extension, the `exec` function is an asynchronous operation that returns a `Promise` object. It executes a
+specific operation defined by an execution type.
+
+```js
+await exec(executionType, ...additionalArguments)
+```
+
+The first argument of the `exec` function is an `executionType` string, which represents the type of operation to be
+conducted. The `executionType` should correspond to a valid, available execution type.
+
+Additional parameters required by the `exec` function depend on the provided execution type. These are added after the
+`executionType` parameter.
+
+There are built-in execution types provided with the `exec` function that are available to all extensions, irrespective
+of its permissions, while other execution types may be added through granted permissions.
+
+:::warning Handle Optional Permissions
+
+If an invalid or non-existent `executionType` is passed to the `exec` function, it will throw an error. If you set some
+permissions as optional, you should handle both cases where they are and are not granted.
+
+:::
+
+## Log
+
+The `log` execution type allows you to show a message in the console as `console.log` is not available in isolates.
```

Comment:
WDYT about such this small rephrase?
```suggestion
The `log` execution type allows you to log a message to the console, as an alternative to `console.log` which is not available in isolates.
```

I would formulate the other descriptions of execution types like this as well: as an alternative to ...
to ensure that no 1-to-1 comparison is made